catalog/src/main/java/org/killbill/billing/catalog/DefaultVersionedCatalog.java

```diff
@@ -107,17 +105,9 @@ private int indexOfVersionForDate(final Date date) throws CatalogApiException {
         if (!versions.isEmpty()) {
             return 0;
         }
-        throw new CatalogApiException(ErrorCode.CAT_NO_CATALOG_FOR_GIVEN_DATE, date.toString());
+        throw new IllegalStateException("No existing versions in the VersionedCatalog catalog ??");
```

Comment:
Maybe add some debugging information (`Date` and/or known versions).

Reply:
ack